proposals/improved-interpolated-strings.md

```diff
@@ -287,12 +270,107 @@ Do we want to have builders for `System.IFormattable` and `System.FormattableStr
+    * If no applicable methods were found, step 3 is retried, removing the final `bool` parameter from `A`. If this retry also finds no applicable members, an error is produced and
+    no further steps are taken.
+    * If no single-best method was found, the result of overload resolution is ambiguous, an error is produced, and no further steps are taken.
+4. Final validation on `F` is performed.
+    * If `F`'s return type is not a `T`, an error is produced and no further steps are taken.
+    * If any element of `A` occurred lexically after `i`, an error is produced and no further steps are taken.
+
+By requiring `Create` to be a `static` method instead of a constructor, we allow the implementation to pool builders if it so decides to.
+If we limited the pattern to constructors, then the implementation would be required to always return new instances.
+
+Additionally, by returning the builder type instead of requiring it to go in an out parameter, we marginally improve the codegen and significantly simplify the rules around
+ref struct lifetimes vs the traditional .NET `TryX` pattern, and we expect a number of these builder types to be ref structs.
 
-Both the general pattern and the specific changes for interpolated strings directly converted to `string`s follow the same lowering pattern. The `Create` method is
-invoked on the receiver (whether that's the temporary method receiver for an _implicit\_string\_builder\_conversion_ derived from the applicable function member algorithm, or a
-standard conversion derived from the target type). If the call returned `true`, `Append...` is repeatedly invoked on the builder out parameter, with each part of the interpolated string,
-in order, stopping subsequent calls if a `Append...` call returns `false`. Finally, the original method is called, passing the initialized builder in place of the interpolated string expression.
+This can and will be impacted by abstract statics in interfaces for generic contexts. We will need to make sure the interactions are considered and tested.
+
+#### `Append...` method overload resolution
+
+Given an _applicable\_interpolated\_string\_builder\_type_ `T` and an _interpolated\_string\_expression_ `i`, overload resolution for a set of valid `Append...` methods on `T` is
+performed as follows:
+
+1. If there are any _interpolated\_regular\_string\_character_ components in `i`:
+    1. Member lookup on `T` with the name `AppendLiteral` is performed. The resulting method group is called `Ml`.
+    2. The argument list `Al` is constructed with one value parameter of type `string`.
+    3. Traditional method invocation resolution is performed with method group `Ml` and argument list `Al`. For the purposes of method invocation final validation, the context of `Ml`
+    is treated as a _member\_access_ through an instance of `T`.
+        * If a single-best method `Fi` is found and no errors were produced, the result of lookup is `Fi`.
```

Comment:
>the result of lookup is `Fi` [](http://example.com/codeflow?start=77&length=28)

"the result of overload resolution is `Fi`"? To match the previous section about the Create method.